tools/pipelines/templates/build-npm-package.yml

```diff
@@ -613,3 +617,6 @@ stages:
         namespace: ${{ parameters.namespace }}
         tagName: ${{ parameters.tagName }}
         publishNonScopedPackages: ${{ variables.publishNonScopedPackages }}
+
+  # Run additional Steps
+  - ${{ parameters.additionalSteps }}
```

Comment:
We said to put this at the end but I realized this file is a bit more complex. There are several stages, and the second one (the build stage) has several jobs. We need this to happen at the end _of the build job inside the build stage_.

![image](https://github.com/microsoft/FluidFramework/assets/716334/dff43b5c-ba1a-4d09-89f0-e0776b837f34)

So between lines 586 and 588. And just make sure that the indentation is correct (in line with all the other steps inside that job). Also for extra clarity, I think let's rename it to `additionalBuildSteps`.

Reply:
Then can I just make a preBuildStep and postBuildStep for inject token and upload artifact? We will still have one time build.

Reply:
Hmm it's tempting, but I think it makes it more complicated because we have to think about exactly when before/after the build we put them. Several other things happen before and after, so the additional steps could try to make assumptions about what has already happened or not. Let's just start with additionalBuildSteps appended to the end of the job, even if it means building that specific package again for now.

Reply:
Oh? Isn't it the step on 285-ish line?

Reply:
Ah. You are right. That will inject .env for every buildable packages. Reverting it

Reply:
The build step is, but if we let the "callers" of this template specify pre-build steps and post-build steps, we would need to figure out if those need to execute immediately before and after the build, or further up (before other of the current pre-build steps in the template execute), and same for the post-ones.

Reply:
Hmm.. Like` ${{ if( eq(workDir, ./packages/tools/devtools/devtools-browser-extension/)}}`?

Reply:
Updated